test/integration/http_integration.cc

```diff
@@ -842,7 +842,7 @@ void HttpIntegrationTest::testRouterUpstreamDisconnectBeforeRequestComplete() {
     EXPECT_EQ(response->headers().getProxyStatusValue(),
               "envoy; error=connection_terminated; "
               "details=\"upstream_reset_before_response_started{connection_termination|QUIC_NO_"
-              "ERROR|Closed_by_application}; UC\"");
+              "ERROR|from:0|Closed_by_application}; UC\"");
```

Comment:
`from:0` is slightly less cool than `FROM_SELF`. Can we call `ConnectionCloseSourceToString()` instead?